api/filters.go

```diff
@@ -446,11 +525,31 @@ func setErrorCode(w http.ResponseWriter, err error, typ ...string) {
 	case filters.ErrFilterOutputNotFound:
 		http.Error(w, err.Error(), http.StatusNotFound)
 		return
-	case errUnauthorised:
+	case filters.ErrUnauthorised:
 		http.Error(w, err.Error(), http.StatusUnauthorized)
 		return
-	default:
-		http.Error(w, internalError, http.StatusInternalServerError)
+	case filters.ErrBadRequest:
+		http.Error(w, badRequest, http.StatusBadRequest)
 		return
+
+	default:
+
+		switch err.(type) {
+		case filters.BadRequestErr:
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		default:
+			http.Error(w, internalError, http.StatusInternalServerError)
+			return
+		}
 	}
 }
+
+func handleAuditingFailure(w http.ResponseWriter, err error, logData log.Data) {
+	logAuditFailure(err, logData)
+	http.Error(w, "internal server error", http.StatusInternalServerError)
+}
+
+func logAuditFailure(err error, logData log.Data) {
+	log.ErrorC("error while attempting to record audit event", err, logData)
```

Comment:
Maybe consider passing in request.Context into this func - you can get the user, caller and request ID from it and add them to the logData.

I won't request changes just for that but something to consider based on our logging chat yesterday.